src/socket.rs

```diff
@@ -156,7 +156,7 @@ impl<A: NetworkAddress, S> Socket<A, S> {
     ) -> std::io::Result<Option<Timestamp>> {
         const TIMEOUT: Duration = Duration::from_millis(200);
 
-        if let Ok(timestamp) = tokio::time::timeout(TIMEOUT, async {
+        let fut = async {
```

Comment:
a 10+ line expression in the if condition... just looks weird and is hard to parse

Reply:
For me, the if expression more clearly shows what the code is doing, instead of having the code and its execution completely separated. I am not saying there ain't ways to improve this, there may well be, but for me this isn't it.